machine/v1beta1/types_gcpprovider.go

```diff
@@ -54,9 +64,21 @@ type GCPMachineProviderSpec struct {
 	// ProjectID is the project in which the GCP machine provider will create the VM.
 	// +optional
 	ProjectID string `json:"projectID,omitempty"`
-	// Preemptible indicates if created instance is preemptible
+	// GPUs is a list of GPUs to be attached to the VM.
+	// +optional
+	GPUs []GCPGPUConfig `json:"GPUs,omitempty"`
+	// Preemptible indicates if created instance is preemptible.
 	// +optional
 	Preemptible bool `json:"preemptible,omitempty"`
+	// OnHostMaintenance determines the behavior when a maintenance event occurs that might cause the instance to reboot.
+	// This is required to be set to TerminateHostMaintenanceType if you want to provision machine with attached GPUs.
+	// +kubebuilder:validation:Enum=MigrateHostMaintenanceType;TerminateHostMaintenanceType;
+	// +optional
+	OnHostMaintenance GCPHostMaintenanceType `json:"onHostMaintenance,omitempty"`
+	// AutomaticRestart determines the behavior when an instance crashes or the underlying infrastructure provider stops the instance as part of a maintenance event (default true).
+	// Cannot be true with preemptible instances.
+	// +optional
+	AutomaticRestart bool `json:"automaticRestart,omitempty"`
```

Comment:
looks like this one slipped through the cracks: https://github.com/openshift/api/pull/1044#discussion_r740695358

Perhaps something like `RestartPolicy`, with possible values of RestartAlways, RemainStopped (or RestartNever for congruency), and ""

Reply:
I do not see much value in changing `bool` to `RestartPolicy`, since it supports only two possible values anyway. It just adds changes to existing code, plus extra need for validation.

I added it as a separate commit, so if you agree we can just remove it.